crawler.py

```diff
@@ -422,7 +565,20 @@ def _scrape_pages(self, aspect_list):
         loop.run_until_complete(asyncio.wait(tasks))
 
 # Async method for actual scraping
+    ":"
+
+
+
+
+    9
     async def scrape_url(self, url, sem):
+        """
+        Asynchronous method that scrape page. Calls save_html() to save scraped page to file.
+        Calls record_milestone() to save progress
+        :param url: url to scrape
+        :param sem: rate limitin semaaphore.
+        :return:
+        """
         async with sem:
             async with aiohttp.ClientSession() as s:
```

Comment:
Hey so there's a problem here that because record_milestone uses the database .task file still instead of json, you get an asyncio Task Exception every time you call record_milestone